### DISCUSSION TITLE
Possible to fetch value from another collection without query?

### DISCUSSION THREAD

**Discussion by CovertCode** [pocketbase/pocketbase#5316] [Q&A]
Env: Nodejss-pocketbase-sdk

```
var initialSearchNumber = 3;
    const instanceList = await pb.collection('instance').getList(0, 50, {
        filter: `status = "running" && currently_used_by = "" && used_by:length >= ${initialSearchNumber}`,
    });
```

Right now I have hardcoded `initialSearchNumber` in my code. I want to store this value in a collection and update the value using pb web UI. So that when my query runs it will fetch value from database without another query.

There's also a way to write query that will fetch this field value from collection and then use it in my `instanceList` query. But I want pb query to directly get value from db without e writing a query.
Something like this:

```
 const instanceList = await pb.collection('instance').getList(0, 50, {
        filter: `status = "running" && currently_used_by = "" && used_by:length >= _db.collection.initialSearchNumber `,
    });
```

is it possible to do something like this?

**Comment by ganigeorgiev** [comment]
I'm confused and not sure that I understand what `initialSearchNumber` and `_db.collection.initialSearchNumber` is supposed to represent and from where they should come from.

In general you can filter `instance` records based on other collection(s) fields only if you have a relation to it (either direct or [indirect/back-relation](https://pocketbase.io/docs/working-with-relations/#back-relations)).

If you need further help, please be more specific about your collections structure (_you could export it as json from the Admin UI > Settings > Export collections_) and elaborate a little more on your use case to understand better the bigger picture and the desired outcome.

**Reply by CovertCode** [reply]
```
var initialSearchNumber = 3;
    const instanceList = await pb.collection('instance').getList(0, 50, {
        filter: `status = "running" && currently_used_by = "" && used_by:length >= ${initialSearchNumber}`,
    });
``` 
in above query I have filter `used_by:length >= ${initialSearchNumber}`
`initialSearchNumber` is a number.
I wanted to reference this value in filter from another collecton.

Here's similar query in sqlite:
```
SELECT *
FROM instance
WHERE status = 'running'
  AND currently_used_by = ''
  AND used_by >= (
      SELECT initial_search_number
      FROM settings
      LIMIT 1
  );
```

Notice this part:
```
AND used_by >= (
      SELECT initial_search_number
      FROM settings
      LIMIT 1
  );
```

**Reply by ganigeorgiev** [reply]
Again, you can filter `instance` records based on other collection(s) fields **only** if you have a relation to it (either direct or [indirect/back-relation](https://pocketbase.io/docs/working-with-relations/#back-relations)).

Without details about your collections structure I don't think I can help further.
